yarn-project/test-wallet/src/wallet/test_wallet.ts

```diff
@@ -161,12 +161,17 @@ export abstract class BaseTestWallet extends BaseWallet {
     executionPayload: ExecutionPayload,
     opts: SimulateMethodOptions,
   ): Promise<TxSimulationResult> {
+    if (this.simulatedSimulations && opts.fee?.estimateGas) {
```

Comment:
This is not necessarily true, but more data is needed (specially when this feature comes online in public). ATM they're never used simultaneously, but I thought for the `TestWallet` this was a reasonable protection.